x/interchainstaking/keeper/ibc_packet_handlers.go

```diff
@@ -492,13 +494,18 @@ func (k *Keeper) HandleWithdrawForUser(ctx sdk.Context, zone *types.Zone, msg *b
 
 	period := int64(k.GetParam(ctx, types.KeyValidatorSetInterval))
 	query := stakingtypes.QueryValidatorsRequest{}
-	return k.EmitValSetQuery(ctx, zone.ConnectionId, zone.ChainId, query, sdkmath.NewInt(period))
+
+	if zone.IsSubzone() {
+		return nil
+	}
```

Comment:
Actually, we probably _do_ want to requery the valset here; because the withdrawal will have changed the state on the hostchain. Just needs directing at zone.ChainID()
```suggestion
```